src/test/java/teammates/test/cases/BaseTestCaseWithBackDoorApiAccess.java

```diff
@@ -10,26 +10,55 @@
 import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
 import teammates.common.datatransfer.attributes.InstructorAttributes;
 import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Const;
 import teammates.test.driver.BackDoor;
+import teammates.test.driver.retry.RetryManager;
+import teammates.test.driver.retry.RetryableTaskReturns;
 
 /**
  * Base class for all test cases which are allowed to access the Datastore via {@link BackDoor}.
  */
 public abstract class BaseTestCaseWithBackDoorApiAccess extends BaseTestCaseWithDatastoreAccess {
 
+    protected AccountAttributes getAccount(String googleId) {
```

Comment:
I feel that the existing abstract methods aren't very useable as they overgeneralise, and do not tell me which fields and which corresponding BackDoor (or BackDoorLogic) method they are using. Some of the entity types have multiple ways of lookup. For example, `getInstructor(InstructorAttributes instructor)` might use `BackDoor.getInstructorByEmail(instructor.email, instructor.courseId)` or `BackDoor.getInstructorByGoogleId(instructor.googleId, instructor.courseId)`.

Instead, I feel that `BaseTestCaseWithBackDoorApiAccess` should mirror the backdoor methods. The specific tests utilising these methods will know which lookup method they need, and call the correct one with the correct fields.